lib/api/core/httpRouter/index.js

```diff
@@ -135,99 +135,101 @@ class Router {
   }
 
   /**
-   * Route an incoming HTTP httpRequest to the right handler
+   * Route an incoming HTTP message to the right handler
    *
-   * @param {object} httpRequest - HTTP httpRequest formatted by Kuzzle Proxy
+   * @param {HttpMessage} message - Parsed HTTP message
    * @param {function} cb
    */
-  route(httpRequest, cb) {
-    let request;
-
-    debug('Routing request: %a', httpRequest);
-
-    if (!this.routes[httpRequest.method]) {
-      // the http protocol uses the requestId as connection id
-      const requestContext = this.kuzzle.router.connections.get(
-        httpRequest.requestId);
-
-      request = new Request(
-        { requestId: httpRequest.requestId },
-        requestContext && requestContext.toJSON());
-      request.response.setHeaders(this.defaultHeaders);
-
-      if (httpRequest.method.toUpperCase() === 'OPTIONS') {
-        request.input.headers = httpRequest.headers;
-        request.setResult({}, 200);
-
-        this.kuzzle.pipe('http:options', request)
-          .then(result => {
-            cb(result);
-            return null;
-          })
-          .catch(error => replyWithError(cb, request, error));
-
-        return;
-      }
-
-      replyWithError(
-        cb,
-        request,
-        errorsManager.getError('unrecognized_http_method', httpRequest.method));
+  route(message, cb) {
+    debug('Routing HTTP message: %a', message);
 
-      return;
+    if (!this.routes[message.method]) {
+      return this.routeUnhandledHttpMethod(message, cb);
     }
 
-    httpRequest.url = httpRequest.url.replace(LeadingSlashRegex, '');
+    message.url = message.url.replace(LeadingSlashRegex, '');
+
+    let request;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Move the declaration of "request" to line 169. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS2392)